### PR TITLE
[tests]migrate test-many-tasks/test-dead-actors to nightly tests

### PR DESCRIPTION
**Pull request by scv119**
This probably doesn't work as I don't have proper credentials set-up yet. But might worth put it up for early feedbacks.
image

### CODE REVIEW

File: release/nightly_tests/stress_tests/many_tasks_compute.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    instance_type: m4.16xlarge
+    resources: {"CPU": 64}
+
+worker_node_types:
+    - name: worker_node
+      instance_type: m4.large
+      min_workers: 100
+      max_workers: 100
+      use_spot: true 
+      resources: {"CPU": 2}
```
Comment (scv119) [inline release/nightly_tests/stress_tests/many_tasks_compute.yaml:21]: GPU: 100

File: release/nightly_tests/stress_tests/test_many_tasks.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        # x_ids = []
+        # for _ in range(50):
+        #     for size_exponent in [0, 1, 2, 3, 4, 5, 6]:
+        #         x_ids = [a.method.remote(10**size_exponent, *x_ids) for a in actors]
+        # ray.get(x_ids)
+        # logger.info("Finished after %s seconds.", time.time() - start_time)
+    except Exception as e:
+        logging.exception(e)
+        success = 0
+    with open(os.environ["TEST_OUTPUT_JSON"], "w") as f:
```
Comment (scv119) [inline release/nightly_tests/stress_tests/test_many_tasks.py:214]: record the stages

**Review by rkooo567 — APPROVED** [review]
LGTM!! Let me add this to the nightly test. Let's discuss with @krfricke how we can run the weekly test

File: release/nightly_tests/stress_tests/stress_tests_compute.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+aws:
+    BlockDeviceMappings:
+        - DeviceName: /dev/sda1
+          Ebs:
+            VolumeSize: 500
+
+head_node_type:
+    name: head_node
+    instance_type: m4.16xlarge
+    resources:
```
Comment (rkooo567) [inline release/nightly_tests/stress_tests/stress_tests_compute.yaml:15]: I remember we needed sth like `resources: 100` before. Why did we remove it?

**Comment by krfricke** [comment]
This looks good, I'll try to kick these off later today!

File: release/nightly_tests/nightly_tests.yaml [hunk 4]
```diff
@@ -227,3 +227,39 @@
   smoke_test:
     cluster:
       compute_template: shuffle/shuffle_compute_smoke.yaml  # Does not exist yet
+
+# Stress tests with many tasks
+- name: stress_test_many_tasks
+  owner:
+    mail: "chenshen@anyscale.com"
+    slack: "@proj-data-processing"
+
```
Comment (krfricke) [inline release/nightly_tests/nightly_tests.yaml:236]:
```suggestion
```

File: release/nightly_tests/nightly_tests.yaml [hunk 5]
```diff
@@ -227,3 +227,39 @@
...
+  smoke_test:
+    cluster:
+      compute_template: shuffle/shuffle_compute_smoke.yaml  # Does not exist yet
+
+# Stress tests with dead actors
+- name: stress_test_dead_actors
+  owner:
+    mail: "chenshen@anyscale.com"
+    slack: "@proj-data-processing"
+
```
Comment (krfricke) [inline release/nightly_tests/nightly_tests.yaml:254]:
```suggestion
```

**Review by krfricke — APPROVED** [review]
Tests pass!

https://buildkite.com/ray-project/periodic-ci/builds/189#b6374142-7523-4312-ae85-c9e1805dc002

One thing Im wondering is if we should just move this to `stress_tests` directory? The `nightly_tests` directory doesn't make much sense and should probably moved anyway.

File: release/nightly_tests/nightly_tests.yaml (hunk 4 above)
Comment (krfricke) [reply release/nightly_tests/nightly_tests.yaml:236]: (we don't keep track of these here anymore)

**Comment by rkooo567** [comment]
@krfricke We are currently aggregating test suites using the nightly_test keyword. Why don't we merge it and rename the whole test suite as Ray_core_tests or sth?

**Comment by rkooo567** [comment]
Test failures should be unrelated

**Comment by rkooo567** [comment]
Also, @krfricke can you set this up running weekly?

**Pull request merged**